clang/include/clang/Sema/Sema.h

```diff
@@ -314,6 +314,7 @@ class SYCLIntegrationHeader {
     kind_accessor = kind_first,
     kind_std_layout,
     kind_sampler,
+    kind_specialization_constants_buffer,
```

Comment:
So quick nit: This ends up causing an API break, doesn't it?  I realize it is pretty academic, as the int-headers are typically generated at the same time, but anyone with an existing int-header will have their code broken thanks to this.

Reply:
Can you explain why adding a new kernel_param_kind_t would cause API break? I suspect I need to handle this in some other areas to make it work correctly before usage, but why would it cause existing integration headers to break?

Reply:
Sure!  Basically the number `4` is currently `kind_pointer`.  If you add an entry in the middle, that `4` compiled in somewhere now means `kind_specialization_constants_buffer` (or, a version of the runtime that is old enough to still have it as pointer works with it differently!).  so now, any attempt to use kind_pointer in any non-completely-contextual way (or again, against a slightly older version of the runtime) now actually ends up referring to the constants buffer.

However, if instead you append to the list, this problem doesn't happen, and the worst you get is an unknown index later.

Reply:
oh that makes sense. Thanks for explaining. I'll make the change.

Reply:
Done!